packages/ui-react/src/containers/AccountModal/AccountModal.tsx

```diff
@@ -97,82 +86,30 @@ const AccountModal = () => {
-      );
+      return fallback;
     }
 
-    switch (view) {
-      case 'login':
-        return <Login />;
-      case 'create-account':
-        return <Registration />;
-      case 'personal-details':
-        return <PersonalDetails />;
-      case 'choose-offer':
-        return <ChooseOffer />;
-      case 'edit-card':
-        return <EditCardDetails />;
-      case 'upgrade-subscription':
-        return <ChooseOffer />;
-      case 'upgrade-subscription-error':
-        return <UpgradeSubscription type="error" onCloseButtonClick={closeHandler} />;
-      case 'upgrade-subscription-success':
-        return <UpgradeSubscription type="success" onCloseButtonClick={closeHandler} />;
-      case 'upgrade-subscription-pending':
-        return <UpgradeSubscription type="pending" onCloseButtonClick={closeHandler} />;
-      case 'checkout':
-        return <Checkout />;
-      case 'payment-error':
-        return <PaymentFailed type="error" message={message} onCloseButtonClick={closeHandler} />;
-      case 'payment-cancelled':
-        return <PaymentFailed type="cancelled" message={message} onCloseButtonClick={closeHandler} />;
-      case 'welcome':
-        return <Welcome onCloseButtonClick={closeHandler} onCountdownCompleted={closeHandler} siteName={siteName} />;
-      case 'reset-password':
-        return <ResetPassword type="reset" />;
-      case 'forgot-password':
-        return <ResetPassword type="forgot" />;
-      case 'add-password':
-        return <EditPassword type="add" />;
-      case 'delete-account':
-      case 'delete-account-confirmation':
-        return <DeleteAccountModal />;
-      case 'warning-account-deletion':
-        return <DeleteAccountPasswordWarning />;
-      case 'send-confirmation':
-        return <ResetPassword type="confirmation" />;
-      case 'edit-password':
-        return <EditPassword />;
-      case 'unsubscribe':
-        return <CancelSubscription />;
-      case 'renew-subscription':
-        return <RenewSubscription />;
-      case 'payment-method':
-      case 'payment-method-success':
-        return <UpdatePaymentMethod onCloseButtonClick={closeHandler} />;
-      case 'waiting-for-payment':
-        return <WaitingForPayment />;
-      case 'finalize-payment':
-        return <FinalizePayment />;
-    }
-  };
+    if (!currentModal) return null;
 
-  const shouldShowBanner = !['delete-account', 'delete-account-confirmation', 'edit-card', 'warning-account-deletion'].includes(view ?? '');
-  const dialogSize = ['delete-account-confirmation'].includes(view ?? '') ? 'large' : 'small';
+    return currentModal.component;
+  };
 
   return (
     <Dialog size={dialogSize} open={!!viewParam} onClose={closeHandler}>
-      {shouldShowBanner && banner && (
+      {!shouldShowBanner && banner && (
         <div className={styles.banner}>
           <img src={banner} alt="" />
         </div>
       )}
-      {renderForm()}
+      <Suspense fallback={fallback}>{renderForm()}</Suspense>
```

Comment:
Minor: we don't need a function anymore. We can just early return above and render currentModal.component.